File: articles/norbert-wiener-cybernetics-and-feedback-in-computation/fortran/negative_feedback.f90
program negative_feedback_example
  implicit none
  integer :: i
  real :: x, target, gain
  x = 10.0
  target = 0.0
  gain = 0.2
  do i = 1, 5
    x = x - gain * (x - target)
  end do
  print *, "final_state=", x
end program negative_feedback_example

File: articles/norbert-wiener-cybernetics-and-feedback-in-computation/rust/negative_feedback.rs
fn main() {
    let mut x = 10.0;
    let target = 0.0;
    let gain = 0.2;
    for _ in 0..5 {
        x = x - gain * (x - target);
    }
    println!("final_state={:.6}", x);
}
